### PR TITLE
Adding python script - PubMed crawler

### PR DESCRIPTION
**Pull request by vpchung**

### CODE REVIEW

**Comment by jaeddy** [comment]
Ah! This notification got buried in my inbox. I'll take a look at the PR tonight or tomorrow.

**Comment by vpchung** [comment]
Oh no worries!  We are all understandably busy this week haha.

Thank you for the help!

File: python/projects_and_cores.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+            print(syn_id)  # diagnostic.
+
+            # U01 grants only has one project and the description is
+            # listed on the main Wiki page.
+            if project.grantType[0] == "U01":
+                project_page_id = project_wiki[0].id
+                project_page = syn.getWiki(project, project_page_id).markdown
+                project_desc = parse_u01_project(project_page)
+                out.write("\t".join([project.centerName[0], "Project", "1",
```
Comment (jaeddy) [inline python/projects_and_cores.py:86]: Can you add a header row as the first thing that gets written to the TSV?

File: python/projects_and_cores.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    """Parse project and core descriptions from U54 center."""
+
+    # Center has only project description. Add '#CORE' to the end so that
+    # all project descriptions can be captured.
+    if not re.search(r"P(ROJECT|roject).*#CORE", page, re.S):
+        page += "\n###COR"
+
+    project_subpage = re.search(r"(Project.*)###COR", page, re.S).group(1)
+    projects = re.split(r"\n####\*", project_subpage)
+    project_info = parse_information(center, projects)
```
Comment (jaeddy) [inline python/projects_and_cores.py]: Minor thing, but probably makes more sense to move the function definition for `parse_information` above `parse_u54_center`, as the former is called by the latter.

File: python/projects_and_cores.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    if cores_found:
+        core_subpage = cores_found.group(2)
+        cores = re.split(r"\n####\*", core_subpage)
+        core_info = parse_information(center, cores)
+
+    return project_info, core_info
+
+
+def parse_information(center, sections):
+    all_info = ""
```
Comment (jaeddy) [inline python/projects_and_cores.py]: Some of the project names are getting parsed as `&nbsp;` — is this expected?

For example:
```
Chicago Region Physical Science Oncology Center	Project	3	&nbsp;	'[Project 3 ...'
H Lee Moffitt Cancer Center and Research Institute	Project	1	&nbsp;	'Cancer is ...'
```

File: python/pubmed_crawler.py [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    """Main function."""
+
+    syn = login()
+    args = get_args()
+
+    # In order to make >3 Entrez requests/sec, 'email' and 'api_key'
+    # params need to be set. Using Milen's info...
+    Entrez.email = 'milen.nikolov@sagebase.org'
+    Entrez.api_key = '3f8cfef8d4356963e36d145c96b9ca9ece09'
+
```
Comment (jaeddy) [inline python/pubmed_crawler.py:338]: Let's use environment variables for credentials, just to be safe. Here's an example:
https://github.com/Sage-Bionetworks/prov-service/blob/develop/synprov/config.py#L25-L26

**Review by jaeddy — CHANGES_REQUESTED** [review]
I haven't ran the pub crawler script yet (though I realize that's the main point of this PR). Seems like there are some issues with the wiki parsing that need a closer look:

image

A few more minor changes mentioned inline. I'll check out the other script soon!

File: python/projects_and_cores.py (hunk 3 above)
Comment (vpchung) [reply python/projects_and_cores.py]: Ah yeah, this occurs when there isn't a title for the project/core.  But I can definitely make it so the title is an empty string instead!

**Comment by vpchung** [comment]
I tried reproducing the output you got but haven't been successful... hopefully with the next commit, things will be okay on your end!

**Comment by jaeddy** [comment]
OK thanks! I'll take another look.

**Comment by jaeddy** [comment]
@vpchung I took a somewhat unconventional approach and created a [PR](https://github.com/vpchung/csbc-pson-dcc/pulls) to add some changes to your branch... which needs to be merged so that those commits are reflected in this PR (I think). 🤔 

After that, we can merge this PR.

In the future, I think it's fine for any of us at Sage to use branches (rather than forks) for updates to this repo. Makes it a bit easier to keep track of things. :)

**Comment by vpchung** [comment]
Ahh, thank you, @jaeddy !

I had attempted to create a branch before but I got a `Permissions denied` when I tried to push..... hence the fork.  But I agree, working off of branches would definitely be easier :) Thanks again!

**Pull request merged**